packages/react/src/components/Callout/CalloutContent.base.tsx

```diff
@@ -111,6 +111,37 @@ function useBounds(
   return getBounds;
 }
 
+/**
+ * (Hook) to return the maximum available height for the Callout to render into.
+ */
+function useMaxHeight(
+  positions: ICalloutPositionedInfo | undefined,
+  { calloutMaxHeight, directionalHint, directionalHintFixed, hidden }: ICalloutProps,
+  getBounds: () => IRectangle | undefined,
+) {
+  const [maxHeight, setMaxHeight] = React.useState<number | undefined>();
+  const { top, bottom } = positions?.elementPosition ?? {};
+
+  React.useEffect(() => {
+    const isForcedInBounds = typeof top === 'number' && typeof bottom === 'number';
+    const { top: topBounds, bottom: bottomBounds } = getBounds() ?? {};
+
+    if (!calloutMaxHeight && !isForcedInBounds && !hidden) {
+      if (typeof top === 'number' && bottomBounds) {
+        setMaxHeight(bottomBounds - top);
+      } else if (typeof bottom === 'number' && typeof topBounds === 'number' && bottomBounds) {
+        setMaxHeight(bottomBounds - topBounds - bottom);
+      }
+    } else if (calloutMaxHeight) {
+      setMaxHeight(calloutMaxHeight);
+    } else {
+      setMaxHeight(undefined);
+    }
+  }, [positions, calloutMaxHeight, getBounds, hidden, directionalHint, directionalHintFixed]);
```

Comment:
nit: Keep this alphabetically ordered

```suggestion
  }, [calloutMaxHeight, directionalHint, directionalHintFixed, getBounds, hidden, positions]);
```